src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+namespace System.Reflection.Emit
+{
+    internal sealed class MethodBuilderImpl : MethodBuilder
+    {
+        private readonly Type _returnType;
+        private readonly Type[]? _parameterTypes;
+        private readonly ModuleBuilderImpl _module;
+        private readonly MethodAttributes _attributes;
+        private readonly string _name;
+        private readonly CallingConventions _callingConventions;
+        private readonly TypeBuilderImpl _declaringType;
+
+        internal MethodBuilderImpl(string name, MethodAttributes attributes, CallingConventions callingConventions, Type? returnType,
+            Type[]? parameterTypes, ModuleBuilderImpl module, TypeBuilderImpl declaringType)
+        {
+            _module = module;
+            _returnType = returnType ?? _module.GetTypeFromCoreAssembly("System.Void"); ;
+            _name = name;
+            _attributes = attributes;
+            _callingConventions = callingConventions;
+            _declaringType = declaringType;
+
+            if (parameterTypes != null)
+            {
+                foreach (Type t in parameterTypes)
+                {
+                    ArgumentNullException.ThrowIfNull(t, nameof(parameterTypes));
+                }
+
+                _parameterTypes = new Type[parameterTypes.Length];
+                Array.Copy(parameterTypes, _parameterTypes, parameterTypes.Length);
```

Comment:
```suggestion
                _parameterTypes = new Type[parameterTypes.Length];
                foreach (int i = 0; i < parameterTypes.Length; i++)
                {
                    ArgumentNullException.ThrowIfNull(_parameterTypes[i] = parameterTypes[i], nameof(parameterTypes));
                }
```
We can validate and copy at the same time.